### PR TITLE
Debugger: Tweaks to memory read/write APIs

### PR DESCRIPTION
**Pull request by unknownbrackets**
This mainly adds ways to read/write in larger chunks, including NUL terminated strings (since that's a common thing.)

Also fixes up some error handling for the other memory APIs, since they were sending two error responses for missing address parameters, etc.  Also needs to validate the game is running so memory isn't nullptr.

-[Unknown]

### CODE REVIEW

File: Common/Data/Encoding/Base64.cpp [hunk 1]
```diff
@@ -33,3 +33,48 @@ std::string Base64Encode(const uint8_t *p, size_t sz) {
 
 	return result;
 }
+
+std::vector<uint8_t> Base64Decode(const char *s, size_t sz) {
+	const uint8_t lookup[256] = {
```
Comment (hrydgard) [inline Common/Data/Encoding/Base64.cpp:38]: static const, so it doesn't try to put this on the stack

File: Core/Debugger/WebSocket/MemorySubscriber.cpp [hunk 2]
```diff
@@ -203,3 +299,37 @@ void WebSocketMemoryWriteU32(DebuggerRequest &req) {
...
+//
+// Parameters:
+//  - address: unsigned integer address for the start of the memory range.
+//  - base64: data to write, encoded as base64 string
+//
+// Response (same event name) with no extra data.
+void WebSocketMemoryWrite(DebuggerRequest &req) {
+	auto memLock = Memory::Lock();
+	if (!currentDebugMIPS->isAlive() || !Memory::IsActive())
+		return req.Fail("CPU not started");
```
Comment (hrydgard) [inline Core/Debugger/WebSocket/MemorySubscriber.cpp:313]: return void value?   valid but unusal.  I'd separate req.Fail and return into two statements (but this does have an attraction... hm).

also this is the only single line `if` without braces in this function

**Review by hrydgard — APPROVED** [review]
Just minor style things, feel free to ignore.

Good stuff.

File: Core/Debugger/WebSocket/MemorySubscriber.cpp (hunk 2 above)
Comment (unknownbrackets) [reply Core/Debugger/WebSocket/MemorySubscriber.cpp:306]: I just removed the other braces, I want to keep the validations clear and succinct, so they don't slow down reading the other code.  Which is also why I prefer the return void.

-[Unknown]

Comment (hrydgard) [reply Core/Debugger/WebSocket/MemorySubscriber.cpp:306]: My eyes react to it with some confusion, but I better get used to it I guess. It is quite neat indeed.

**Pull request merged**